src/pages/Payment/PaymentPage.jsx

```diff
@@ -0,0 +1,72 @@
+// PaymentPage.jsx
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { supabaseClient, lakshya } from "../../utils/supabase_helper";
+import useCart from "../../hooks/useCart";
+import ValidatePayment from "../../components/ValidatePayment/ValidatePayment";
+
+const PaymentPage = () => {
+  const navigate = useNavigate();
+  const { cart, clearCart } = useCart();
+  const [loading, setLoading] = useState(true);
+  const [paymentResponse, setPaymentResponse] = useState();
+
+  useEffect(() => {
+    const handlePayment = async () => {
+      if (cart.length === 0) {
+        console.error("Cart is empty. Redirecting to home page.");
+        navigate("/");
+        return;
+      }
+      const userId = await (await supabaseClient.auth.getUser()).data.user.id;
+
+      const paymentEndpoint = `${
+        import.meta.env.VITE_PAYMENT_URL
+      }?user_id=${userId}`;
+      const eventIds = cart.map((event) => event.id);
+      console.log(paymentEndpoint);
+      console.log(eventIds);
+
+      try {
+        // Send payment request
+        const paymentRes = await lakshya.requestPayment(
+          eventIds,
```

Comment:
As per the [docs ](https://yogesh-dubey-ayesavi.github.io/lakshya-sdk/#md:requestpayment) a list of `LakshyaEvent` has to be passed instead a list of eventIds is being passed directly, correcting it will make it work absolutely fine